logstash-core/src/test/java/org/logstash/common/AbstractDeadLetterQueueWriterExtTest.java

```diff
@@ -0,0 +1,79 @@
+package org.logstash.common;
+
+import org.jruby.RubyString;
+import org.jruby.javasupport.JavaUtil;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.logstash.DLQEntry;
+import org.logstash.RubyTestBase;
+import org.logstash.RubyUtil;
+import org.logstash.common.io.DeadLetterQueueReader;
+import org.logstash.common.io.DeadLetterQueueWriter;
+import org.logstash.common.io.QueueStorageType;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Duration;
+
+public class AbstractDeadLetterQueueWriterExtTest extends RubyTestBase {
+
+    static final int DLQ_VERSION_SIZE = 1;
+
+    @Rule
+    public TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    @Test
+    public void testDLQWriterDoesntInvertPluginIdAndPluginTypeAttributes() throws IOException, InterruptedException {
+        final Path dlqPath = tmpFolder.newFolder("dead_letter_queue").toPath();
+        final String pluginId = "id_value";
+        final String pluginType = "elasticsearch";
+        final String dlqName = "main";
+
+        writeAnEventIntoDLQ(dlqPath, pluginId, pluginType, dlqName);
+
+        String segment = "1.log";
+        DeadLetterQueueReader dlqReader = openDeadLetterQueueReader(dlqPath, dlqName, segment);
+
+        DLQEntry dlqEntry = dlqReader.pollEntry(1_000);
+        assertNotNull("A DLQEntry must be present in the queue", dlqEntry);
+        assertEquals(pluginId, dlqEntry.getPluginId());
+        assertEquals(pluginType, dlqEntry.getPluginType());
+    }
+
+    private DeadLetterQueueReader openDeadLetterQueueReader(Path dlqPath, String dlqName, String segment) throws IOException {
+        DeadLetterQueueReader dlqReader = new DeadLetterQueueReader(dlqPath);
+        Path currentSegmentFile = dlqPath.resolve(dlqName).resolve(segment);
+        dlqReader.setCurrentReaderAndPosition(currentSegmentFile, DLQ_VERSION_SIZE);
+        assertEquals(currentSegmentFile, dlqReader.getCurrentSegment());
+        return dlqReader;
+    }
+
+    private void writeAnEventIntoDLQ(Path dlqPath, String pluginId, String pluginType, String dlqName) {
```

Comment:
I am wondering if other tests also write an event to DLQ and could make this method to a util class

Reply:
`DeadLetterQueueWriterFactory` is used in its test https://github.com/elastic/logstash/blob/main/logstash-core/src/test/java/org/logstash/common/DeadLetterQueueFactoryTest.java but I haven't found test where a `AbstractDeadLetterQueueWriterExt` is created and initialized with an instance of the real DLQ writer.